### ISSUE TITLE
Question on the parameters representation in the example of Bayesian NNs

### ISSUE THREAD

**Issue by Xihaier** [thu-ml/zhusuan#66]
Best greetings, dear authors:

I truly appreciate this package and I am playing with one of the examples (Bayesian NNs).

Honestly speaking, I am not working in this specific field and have limited background on this topic. But I would really appreciate that if you could tell the physical meaning of some evaluation parameters in your provided code (bayesian_nn.py).

Starts from line 86 and ends at line 95, I am wondering what's the role of "lb_samples, ll_samples, epochs, test_freq, anneal_lr_freq, anneal_lr_rate".

Hope it's not much work and thanks for your time!

**Comment by thjashin** [comment]
@Xihaier Hi, Thanks for the question. lb_samples represents how many samples are used to estimate the lower bound, ll_samples represents how many samples are used for importance sampling to estimate the log likelihood. epochs is the number of passes through the training set, anneal_lr_freq and anneal_lr_rate is for anneal the learning rate.

**Comment by Xihaier** [comment]
Hi Jiaxin,

It's very helpful! 

If I may, could you give me some suggestions on the selection of a set of appropriate setting of the aforementioned parameters? Say if my input data set X is an N by M matrix, with N denoting the number of samples and M denoting the number of features. Y is an N by 1 vector. I am wondering if there are any tricks in the initialization of these parameters in accordance with my dataset X and Y. If not, I am more than willing to go with trial and error method.

**Comment by thjashin** [comment]
@Xihaier In fact for Bayesian NNs we often normalize the input features to be zero mean and unit standard deviation. And for lb_samples, the principle is that more samples you use, you will get a more accurate estimation of the lower_bound. For ll_samples, it relates to the evaluation for test log likelihood, you will typically need hundreds or thousands of them to obtain a reasonable estimate. The other two (epochs, anneal..) are just parameters for every machine learning program with gradient descent. You can tune it just like you do elsewhere.

**Comment by Xihaier** [comment]
Thanks a lot, I really appreciate your kind patience and helpful instruction.

**Comment by thjashin** [comment]
You are welcome :)
